restic/cli/restore.go

```diff
@@ -265,8 +277,9 @@ func (r *Restic) s3Restore(log logr.Logger, snapshot Snapshot, stats *RestoreSta
 
 	cmd := NewCommand(r.ctx, log, opts)
 	cmd.Run()
-	defer log.Info("restore finished")
+}
 
+func (r *Restic) closeRestoreWriters(finalWriter io.WriteCloser, zipWriter *gzip.Writer, uploadWritePipe *io.PipeWriter) (err error) {
 	// We need to close the writers in a specific order
 	err = finalWriter.Close()
```

Comment:
I would make this less generic too: Remove the `finalWriter` and directly check `tarWriter != nil`. This would make the code less confusing at least for me.